### DISCUSSION TITLE
Adding new DataArray.plot method for 2D DataArrays.

### DISCUSSION THREAD

**Discussion by vasilisniaouris** [pydata/xarray#7852] [Ideas]
Hello all! New xarray and really loving it! More often than not, When I am plotting 2D data arrays, I want to plot lines with an x-axis being one of the array dimensions, and the color of the line being the second dimension. Currently, this can be achieved by `xarray.DataArray.plot.line` by providing `x` and `hue`, and then creating a custom colormap and colorbar. What I am proposing looks a lot like the second example plot on the [matplotlib line_collection page](https://matplotlib.org/stable/gallery/shapes_and_collections/line_collection.html).

![Shows an example of parallel lines that follow the viridis colormaping](https://matplotlib.org/stable/_images/sphx_glr_line_collection_002_2_0x.png)

**Comment by Illviljan** [comment]
See #7173. I think I got stuck whether it should be a new method or not.

**Reply by vasilisniaouris** [reply]
Hello, thank you for your response! I looked at [#7173](https://github.com/pydata/xarray/pull/7173). It seems like it would enable the feature I am interested in. I believe the functionality I am looking for can be added to the `line`  method, by allowing a cmap creation, instead of circling over the matplotlib base colors, which is what you have done in [#7173](https://github.com/pydata/xarray/pull/7173). I was not aware of it. 

I am fairly new to github discussions, so I am not sure if the [#7173](https://github.com/pydata/xarray/pull/7173) has been successfully implemented yet. I tried to run the code of the example and it failed. Is there a timeline as to when it would be implemented?
